src/NServiceBus.Core/Routing/UnicastRoutingTable.cs

```diff
@@ -46,24 +57,21 @@ public void AddStatic(Type messageType, EndpointInstanceName destination)
         /// <param name="destinationAddress">Destination endpoint instance address.</param>
         public void AddStatic(Type messageType, string destinationAddress)
         {
-            rules.Add((t, c) => StaticRule(t, messageType, new UnicastRoute(destinationAddress)));
+            staticRules.Add((t, c) => StaticRule(t, messageType, new UnicastRoute(destinationAddress)));
         }
 
         /// <summary>
         /// Adds a rule for generating unicast routes.
         /// </summary>
-        /// <param name="dynamicRule">The rule.</param>
-        public void AddDynamic(Func<Type, ContextBag, IEnumerable<IUnicastRoute>> dynamicRule)
+       // /// <param name="dynamicRule">The rule.</param>
+        public void AddDynamic(Func<Type, ContextBag, Task<List<IUnicastRoute>>> dynamicRule)
```

Comment:
Not sure I understand here. We are accepting a Func that returns Task<List<IUnicastRoute>>>, so we are not accepting a Task at all, we are accepting a function that returns a task.

Also.. we WANT to execute the rules every time because subscribers might get updated at any time. (We may want to layer on some caching eventually, but we do not want just to call the operation once.)